src/workerd/io/hibernation-manager.c++

```diff
@@ -164,6 +185,42 @@ kj::Promise<void> HibernationManagerImpl::readLoop(HibernatableWebSocket& hib) {
   while (true) {
     kj::WebSocket::Message message = co_await ws.receive();
     // Note that errors are handled by the callee of `readLoop`, since we throw from `receive()`.
+
+    auto skip = false;
+
+    KJ_IF_MAYBE (reqResp, autoResponsePair) {
+      KJ_SWITCH_ONEOF(message) {
+        KJ_CASE_ONEOF(text, kj::String) {
+          if (text == (*reqResp)->getRequest()) {
+            // If the received message matches the one set for auto-response, we must
+            // short-circuit readLoop, store the current timestamp and and automatically respond
+            // with the expected response.
+            TimerChannel& timerChannel = KJ_REQUIRE_NONNULL(timer);
+            // We should have set the timerChannel previously in the hibernation manager.
+            // If we haven't, we aren't able to get the current time.
+            hib.autoResponseTimestamp = timerChannel.now();
+            // We'll store the current timestamp in the HibernatableWebSocket to assure it gets
+            // stored even if the WebSocket is currently hibernating. In that scenario, the timestamp
+            // value will be loaded into the WebSocket during unhibernation.
+            KJ_IF_MAYBE(active, hib.activeOrPackage.tryGet<jsg::Ref<api::WebSocket>>()) {
+              // If the actor is not hibernated/If the WebSocket is active, we need to update
+              // autoResponseTimestamp on the active websocket.
+              (*active)->setAutoResponseTimestamp(hib.autoResponseTimestamp);
+            }
+            ws.send((*reqResp)->getResponse().asArray());
+            skip = true;
```

Comment:
nit: Can we just `continue` here, rather than when we leave this scope?

Reply:
Wierdly, no. 
It'll just skip any other `KJ_CASE_ONEOF`, that way I had the flag and `continue` in the `while` loop.